### DISCUSSION TITLE
Setting boundary conditions using the Fortran interface

### DISCUSSION THREAD

**Discussion by Mephiiistopheles** [AMReX-Codes/amrex#3310] [General]
Are there any tutorials that demonstrate how different boundary conditions (e.g. Dirichlet) can be set via the Fortran interface? I'm just getting start with AMReX and thought that modifying the Advection_octree_F tutorial example to use different boundary conditions would be a good start. The basic idea was to replace the periodic conditions with a 5 m/s velocity condition at the -x boundary, outflow at the +x boundary and slip/non-slip conditions at the -y and +y boundaries such that the 'blob' is advected out of the domain.

I found that this was straightforward enough in C++, but seem to have run into a wall with the Fortran version of the tutorial (which is unfortunate, because I ultimately plan to use this interface due to legacy code). It seems that the fill_physbc subroutine needs to be modified for this case, but I'd greatly appreciate any tips/guidance on this one.

**Comment by WeiqunZhang** [comment]
Maybe you can look at Tests/FortranInterface/Advection_F/Source/fillpatch_mod.F90 as an example. To set velocity at -x boundary, it can be something like below.  (Note that it may not compile, because I did not test.)
```
--- a/Tests/FortranInterface/Advection_F/Source/fillpatch_mod.F90
+++ b/Tests/FortranInterface/Advection_F/Source/fillpatch_mod.F90
@@ -92,6 +92,11 @@ contains
 
              ! amrex_filcc doesn't fill EXT_DIR (see amrex_bc_types_module for a list of bc types
              ! In that case, the user needs to fill it.
+
+             ! -x boundry
+             if (plo(1) <= geom%domain%lo(1)) then
+                p(:geom%domain%lo(1),:,:,velocity_component) = 5.
+             endif
           end if
        end do
        !$omp end parallel
```

**Reply by Mephiiistopheles** [reply]
This is exactly what I needed. Thank you!
